src/lightning_app/core/flow.py

```diff
@@ -177,6 +178,9 @@ def __setattr__(self, name, value):
                 value.component_name = self.name
                 self._state.add(name)
 
+            elif isinstance(value, CloudCompute):
+                self._state.add(name)
```

Comment:
@tchaton can you tell me what happens when
```
        self.other_name = CloudCompute(name="gpu")
```
because until now, we had a special property setter/getter. 
Maybe integrate this into your test. A user might do:

```
self.shared_compute = CloudCompute(name="gpu")
```
and in their run method use it to dynamically create work on the same compute.

Reply:
This works for the flow but not for the works.